mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java

```diff
@@ -121,23 +121,20 @@ public MessageParser() {
         return Stream.empty();
     }
 
-    private MessageAttachment retrieveAttachment(Entity entity) throws IOException {
+    private ParsedAttachment retrieveAttachment(Entity entity) throws IOException {
         Optional<ContentTypeField> contentTypeField = getContentTypeField(entity);
         Optional<ContentDispositionField> contentDispositionField = getContentDispositionField(entity);
         Optional<String> contentType = contentType(contentTypeField);
         Optional<String> name = name(contentTypeField, contentDispositionField);
         Optional<Cid> cid = cid(readHeader(entity, CONTENT_ID, ContentIdField.class));
         boolean isInline = isInline(readHeader(entity, CONTENT_DISPOSITION, ContentDispositionField.class)) && cid.isPresent();
 
-        return MessageAttachment.builder()
-                .attachment(Attachment.builder()
-                    .bytes(getBytes(entity.getBody()))
-                    .type(contentType.orElse(DEFAULT_CONTENT_TYPE))
-                    .build())
-                .name(name.orElse(null))
-                .cid(cid.orElse(null))
-                .isInline(isInline)
-                .build();
+        return ParsedAttachment.builder()
+                .contentType(contentType.orElse(DEFAULT_CONTENT_TYPE))
+                .content(getContent(entity.getBody()))
```

Comment:
well, it looks like we actually keep a byte array hidden in an inputstream. Do we?

Reply:
Yes but mime4j APIs are crappy...